crates/oxc_ecmascript/src/side_effects/may_have_side_effects.rs

```diff
@@ -200,6 +219,56 @@ impl MayHaveSideEffects for BinaryExpression<'_> {
     }
 }
 
+/// Whether the name matches any known global constructors.
+///
+/// <https://tc39.es/ecma262/multipage/global-object.html#sec-constructor-properties-of-the-global-object>
+fn is_known_global_constructor(name: &str) -> bool {
```

Comment:
FYI we have globals saved in https://github.com/oxc-project/oxc/blob/main/crates/oxc_linter/src/javascript_globals.rs

I can move it to its own repo and crate.

Reply:
We need more information than whether it's global. For example, `globalThis` and `JSON` has to be excluded because `1 instanceof globalThis` / `1 instanceof JSON` throws an error because these are not a function.